snorkel/labeling/lf/nlp_spark.py

```diff
@@ -0,0 +1,106 @@
+from snorkel.preprocess.nlp import SpacyPreprocessor
+from snorkel.preprocess.spark import make_spark_preprocessor
+
+from .nlp import (
+    BaseNLPLabelingFunction,
+    SpacyPreprocessorParameters,
+    base_nlp_labeling_function,
+)
+
+
+class SparkNLPLabelingFunction(BaseNLPLabelingFunction):
+    r"""Special labeling function type for SpaCy-based LFs running on Spark.
+
+    This class is a Spark-compatabile version of ``NLPLabelingFunction``.
+    See ``NLPLabelingFunction`` for details.
+
+    Parameters
+    ----------
+    name
+        Name of the LF
+    f
+        Function that implements the core LF logic
+    resources
+        Labeling resources passed in to ``f`` via ``kwargs``
+    pre
+        Preprocessors to run before SpacyPreprocessor is executed
+    fault_tolerant
+        Output -1 if LF execution fails?
+    text_field
+        Name of data point text field to input
+    doc_field
+        Name of data point field to output parsed document to
+    language
+        SpaCy model to load
+        See https://spacy.io/usage/models#usage
+    disable
+        List of pipeline components to disable
+        See https://spacy.io/usage/processing-pipelines#disabling
+    memoize
+        Memoize preprocessor outputs?
+
+    Raises
+    ------
+    ValueError
+        Calling incorrectly defined preprocessors
+
+    Attributes
+    ----------
+    name
+        See above
+    fault_tolerant
+        See above
+    """
+
+    @classmethod
+    def _create_preprocessor(
+        cls, parameters: SpacyPreprocessorParameters
+    ) -> SpacyPreprocessor:
+        preprocessor = SpacyPreprocessor(**parameters._asdict())
+        make_spark_preprocessor(preprocessor)
```

Comment:
The name make_spark_preprocessor makes it sound like it returns a new preprocessor, rather than modifying the given one in place. In this case, it does both. But it still seems more readable to 
```
 preprocessor = write make_spark_preprocessor(preprocessor)
```
Or even better, to rename make_spark_preprocessor to something like make_spark_compatible

Reply:
`make_spark_preprocessor` monkey patches then returns the preprocessor, so the syntax you wrote is valid. but mypy has trouble since `make_spark_preprocessor` has a return type of `Mapper`, so for this internal version, it makes sense to do it like this.